OISF/suricata
# Pcre2 3194 v9

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- Use PCRE2 instead of pcre for transform pcrexform
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature
- transforms : use generic `Free` functions everywhere
- migrate detect-pcre to pcre2
- migrate some pcre1 code to pcre2

After that, the goal is to remove pcre everywhere else cf `git grep pcre_compile`

suricata-verify-pr: 480

https://github.com/OISF/suricata-verify/pull/480

Replaces https://github.com/OISF/suricata/pull/6043 
- migrate more to pcre2
- fix compilation
